vortex-expr/src/scope.rs

```diff
@@ -27,6 +27,35 @@ impl FromStr for Identifier {
     }
 }
 
+impl PartialEq<str> for Identifier {
+    fn eq(&self, other: &str) -> bool {
+        match self {
+            Identifier::Identity => other.is_empty(),
+            Identifier::Other(str) => str.as_ref() == other,
+        }
+    }
+}
+
+impl From<&str> for Identifier {
+    fn from(value: &str) -> Self {
+        if value.is_empty() {
+            Identifier::Identity
+        } else {
+            Identifier::Other(Arc::from(value))
+        }
+    }
+}
+
+impl From<&Arc<str>> for Identifier {
```

Comment:
I'm not really a Rustacean yet but it seems a bit odd to me to define these two `From` implementations on references to values when they both require ownership (which they do via cloning).

Reply:
hmm, you would prefer From<Arc<str>> (and clone) on the outside

Reply:
If this is the style of Vortex let’s stick with it. I think my argument for caller clones is that when you write new code you see the clones and reckon with their effect.